### PR TITLE
set SCL env variables on build

### PR DESCRIPTION
**Pull request by fila43**

### CODE REVIEW

**Comment by fila43** [comment]
[test]

**Comment by fila43** [comment]
[test]

File: src/Dockerfile.template [hunk 1]
```diff
@@ -31,6 +35,26 @@ ENV BASH_ENV=${APP_ROOT}/etc/scl_enable \
 {% endif %}
 
 {{ macros.env_metadata() -}}
+{% if spec.el_version == '7' %}
+ENV SUMMARY="Platform for building and running Python $PYTHON_VERSION applications" \
```
Comment (frenzymadness) [inline src/Dockerfile.template:39]: Šlo by summary a description udělat tak, aby nebyly v šabloně duplicitně?

**Review by frenzymadness — COMMENTED** [review]
Testy neprocházejí, ale ještě nevím proč. Nicméně podezřelé hlášky jsou:
```
test/run: line 12: warning: command substitution: ignored null byte in input
[INFO] Testing 's2i usage' ...
test/run: line 13: warning: command substitution: ignored null byte in input
```

File: src/Dockerfile.template (hunk 1 above)
Comment (fila43) [reply src/Dockerfile.template:39]: ok zkusim to, ty testy už vím o co jde opravim to zaroven

**Comment by fila43** [comment]
[test]

**Comment by fila43** [comment]
[test]

**Comment by frenzymadness** [comment]
Could we make this PR a draft until it's ready for review?

**Comment by fila43** [comment]
[test]

**Comment by fila43** [comment]
[test]

**Comment by fila43** [comment]
[test]

**Comment by fila43** [comment]
[test]

**Comment by fila43** [comment]
rhel8 test fails (also in ruby container) probably due to podman issue, but this PR focuses on rhel7 and centos7 containers,so it should affect only rhel7 and centos7 containers, base on this PR should be ready for review

**Comment by fila43** [comment]
[test]

File: 2.7/Dockerfile [hunk 2]
```diff
@@ -5,7 +5,16 @@ FROM centos/s2i-base-centos7
 EXPOSE 8080
 
 ENV PYTHON_VERSION=2.7 \
-    PATH=$HOME/.local/bin/:$PATH \
+    PYTHON_SCL_VERSION=27 \
+     PATH=$PATH:/opt/rh/rh-python27/root/usr/bin:$HOME/.local/bin/:/opt/rh/$NODEJS_SCL/root/usr/bin:/opt/rh/httpd24/root/usr/bin:/opt/rh/python27/root/usr/bin:/opt/rh/httpd24/root/usr/sbin:/opt/rh/rh-python27/root/usr/local/bin \
```
Comment (frenzymadness) [inline 2.7/Dockerfile:9]: This line and lines below have one more space in the indentation.

**Review by frenzymadness — CHANGES_REQUESTED** [review]
This looks very promising. Please fix the indentation I've commented on and rebase it on top of master. Python 3.6 has been recently disabled for RHEL/Centos 7 so this PR should not generate these Docker files.

**Comment by fila43** [comment]
[test]

**Review by frenzymadness — COMMENTED** [review]
The code itself looks perfect now, thank you!

One last thing before we merge this. Could you please take a look at the errors in tests for RHEL and Cetos 7?

RHEL:
```
[1m[INFO] Testing variable presence during `docker exec`[0m
 All values present in `docker exec`
test/run: line 258: ￼: command not found
[1m[INFO] Checking if all scl variables are defined in Dockerfile[0m
Unable to open /etc/scl/conf/rh-pyton38!
 All scl_enable values present
```

Centos:
```
10:41:56 [1m[INFO] Testing variable presence during `docker exec`[0m
10:41:56 
10:41:59  All values present in `docker exec`
10:41:59 test/run: line 258: $'\357\277\274': command not found
10:41:59 
10:41:59 [1m[INFO] Checking if all scl variables are defined in Dockerfile[0m
10:41:59 
10:42:02 Unable to open /etc/scl/conf/rh-pyton38!
10:42:02  All scl_enable values present
```

**Comment by fila43** [comment]
[test]

**Comment by fila43** [comment]
> The code itself looks perfect now, thank you!
> 
> One last thing before we merge this. Could you please take a look at the errors in tests for RHEL and Cetos 7?
> 
> RHEL:
> 
> ```
> [1m[INFO] Testing variable presence during `docker exec`�[0m
>  All values present in `docker exec`
> test/run: line 258: ￼: command not found
> �[1m[INFO] Checking if all scl variables are defined in Dockerfile�[0m
> Unable to open /etc/scl/conf/rh-pyton38!
>  All scl_enable values present
> ```
> 
> Centos:
> 
> ```
> 10:41:56 �[1m[INFO] Testing variable presence during `docker exec`�[0m
> 10:41:56 
> 10:41:59  All values present in `docker exec`
> 10:41:59 test/run: line 258: $'\357\277\274': command not found
> 10:41:59 
> 10:41:59 �[1m[INFO] Checking if all scl variables are defined in Dockerfile�[0m
> 10:41:59 
> 10:42:02 Unable to open /etc/scl/conf/rh-pyton38!
> 10:42:02  All scl_enable values present
> ```

it should be fixed

**Review by frenzymadness — APPROVED** [review]
This is perfect now, thank you!

**Pull request merged**